src/server/cluster/cluster_defs.cc

```diff
@@ -13,6 +16,30 @@ using namespace std;
 ABSL_FLAG(string, cluster_mode, "", "Cluster mode supported. Default: \"\"");
 
 namespace dfly::cluster {
+std::string SlotRange::ToString() const {
+  return absl::StrCat("[", start, ", ", end, "]");
+}
+
+SlotRanges::SlotRanges(std::vector<SlotRange> ranges) : ranges_(std::move(ranges)) {
+  std::sort(ranges_.begin(), ranges_.end());
+}
+
+void SlotRanges::Extend(const SlotRanges& sr) {
+  ranges_.insert(ranges_.end(), sr.ranges_.begin(), sr.ranges_.end());
+  std::sort(ranges_.begin(), ranges_.end());
```

Comment:
How about using `SlotSet` here instead, and then converting back to `SlotRanges`?
This will have the benefit of squashing adjacent ranges (i.e. `[0, 10], [11, 20]` becomes `[0, 20]`)

Reply:
I thought about that but if we have 1000 nodes we need 1000 SlotSet. it's 16MB of memory. Is it OK?

Reply:
My suggestion is to limit the scope of the `SlotSet` to this method (not a member), so it's only (very) temporary.
I think that with a 1000 node setup we can be ok with that :rofl: